### PR TITLE
:sparkles: DOP-4240 implements filtering sensitive values method to prevent sensitive values being logged

### PR DESCRIPTION
**Pull request by caesarbell**
### Stories/Links:

DOP-4240

### Notes

**Context**: The code changes provided add a filtering method that looks for sensitive values from our param store, and redacts any sensitive values found in the message that is being logged.

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

File: src/enhanced/utils/get-sensitive-values.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+       * Hijacks the getSecureString to populate the
+       * sensitive key array, used in the filterSensitiveValue
+       */
+      sensitiveKeys.push(secureString);
+
+      const envName = paramToEnvMap.get(paramName);
+
+      if (!envName) {
+        console.error(
+          `ERROR! The param '${paramName}' does not having a mapping to an environment variable name. Please define this in the ${resourceName} map.`
```
Comment (rayangler) [inline src/enhanced/utils/get-sensitive-values.ts:91]: (Super nit) I know this was copied over, but I think there's a typo: "having" should be "have"?

**Review by rayangler — APPROVED** [review]
Changes LGTM! I would wait to merge until Brandon reviews, in case he has any important insight

File: src/enhanced/utils/get-sensitive-values.ts (hunk 1 above)
Comment (caesarbell) [reply src/enhanced/utils/get-sensitive-values.ts:91]: @rayangler good catch. I will update that.

Comment (caesarbell) [reply src/enhanced/utils/get-sensitive-values.ts]: updated

**Review by branberry — APPROVED** [review]
LGTM!

**Pull request merged**